disasm/disasm.cc

```diff
@@ -1244,6 +1274,112 @@ disassembler_t::disassembler_t(int xlen)
     }
   }
 
+#define DEFINE_PITYTPE(code, immbti) DISASM_INSN(#code, code, 0, {&xrd, &xrs1, &p_imm##immbti});
+#define DEFINE_ONEOP(code) DISASM_INSN(#code, code, 0, {&xrd, &xrs1});
+
+#define DISASM_8_AND_16_RINSN(code) \
+  DEFINE_RTYPE(code##8); \
+  DEFINE_RTYPE(code##16);
+
+#define DISASM_8_AND_16_RINSN_ROUND(code) \
+  DISASM_INSN(#code "8.u", code##8_u, 0, {&xrd, &xrs1, &xrs2}); \
+  DISASM_INSN(#code "16.u", code##16_u, 0, {&xrd, &xrs1, &xrs2}); \
+
+#define DISASM_8_AND_16_PIINSN(code) \
+  DEFINE_PITYTPE(code##8, 3); \
+  DEFINE_PITYTPE(code##16, 4);
+
+#define DISASM_8_AND_16_PIINSN_ROUND(code) \
+  DISASM_INSN(#code "8.u", code##8_u, 0, {&xrd, &xrs1, &p_imm3}); \
+  DISASM_INSN(#code "16.u", code##16_u, 0, {&xrd, &xrs1, &p_imm4});
+
+  DISASM_8_AND_16_RINSN(add);
```

Comment:
Hi,
     I am trying to add the support for add8 in spike. I can see that it is already updated.
This is the code that I am trying to run:
```
#include <stdio.h>
#include <stddef.h>
#include <stdint.h>

int main() {
    int32_t a[4]={1,2,3,4};
    int32_t b[4]={1,2,3,4};
    int32_t c[4];
    int i;
    
    for (i=0 ; i < 4; i++){
    asm volatile
    (
    "add8   %[z], %[x], %[y]\n\t"
    : [z] "=r" (c[i])
    : [x] "r" (a[i]), [y] "r" (b[i])
    ); 
        //c[i] = a[i] + b[i];
        
        printf("result_vector:%d\n",c[i]);
    }
    return 0;
}
```
I have already added the support for add8 in riscv-gnu -toolchain with updated match and mask bit. The code is properly compiled with riscv32-unknown-elf-gcc.
#define MATCH_ADD8 0x4800007f
#define MASK_ADD8 0xfe00707f

But when I run the executable in spike I get the below error:
```
bl loader
z  00000000 ra 000100bc sp 7fbeed20 gp 00020a90
tp 00000000 t0 00010286 t1 0000000f t2 00000000
s0 7fbeed70 s1 00000000 a0 00000001 a1 7fbeed74
a2 00000001 a3 7fbeed60 a4 00000001 a5 00000001
a6 00000000 a7 00000000 s2 00000000 s3 00000000
s4 00000000 s5 00000000 s6 00000000 s7 00000000
s8 00000000 s9 00000000 sA 00000000 sB 00000000
t3 00000000 t4 00000000 t5 00000000 t6 00000000
pc 00010178 va 48f70777 insn       48f70777 sr 80046020
An illegal instruction was executed!
```
I an running spike after building it in p-ext branch. 
I have a doubt regarding the Define type in disasm.cc. I do not see the instrcutiono name add8 in the updated disasm.cc. Please suggest how to fix the issue.

Reply:
Hi,
      The mask and match bit has been updated as per the update in encoding.h in the p-ext branch.
#define MATCH_ADD8 0x48000077
#define MASK_ADD8 0xfe00707f